rpc/types/src/json.rs

```diff
@@ -635,7 +635,9 @@ define_request_and_response! {
     AccessResponseBase {
         blob: String,
         block_header: BlockHeader,
-        json: String, // FIXME: this should be defined in a struct, it has many fields.
+        /// `cuprate_rpc_types::json::block::Block` should be used
+        /// to create this JSON string in a type-safe manner.
+        json: String,
```

Comment:
I thought we would be replacing embedded JSON within RPC responses with these types but I realized: it's not actual JSON, it's a string promising it contains JSON.

I.e. the RPC response is:
```json
{
  "json": "{\n  \"version\": 1, \n  \"unlock_time\": 0}"
}
```
not:
```json
{
  "json": { "version": 1, "unlock_time": 0 }
}
```

I don't think there's any reason it shouldn't be actual JSON other than some internal `monerod` detail making that hard to do. Anyway, we have to keep this a `String` but we can still use the JSON types to be more type-safe.

Reply:
Or we could add:
- `#[serde(serialize_with = "type_to_string")]`
- `#[serde(deserialize_with = "string_to_type")]`

to automatically do this extra step?

Reply:
It would probably be easier to just keep this as a `String`